dl.ts

```
@@ -191,8 +191,10 @@ export class OpsDL implements types.BackendOps {
     return new TensorDL(ndarray, x.math);
   }
 
-  eye(size: number, dtype: types.DType = "float32"): types.BasicTensor {
-    throw new Error("Not Implemented");
+  setDiag(x: TensorDL, diag: TensorDL): TensorDL {
+    ENV.setMath(x.math);
+    const nd = x.math.setDiag(x.ndarray as Array2D, diag.ndarray as Array1D);
```

What happens when `x` or `diag` don't have the expected rank?

---

It probably blows up. I'd like to leave this as undefined behavior for now, because the TF side [supports arbitrarily ranked inputs](https://github.com/tensorflow/tensorflow/blob/86c10063c8521bd5482df13676bac3575540d9e2/tensorflow/core/ops/ops.pbtxt#L12475-L12493) and I suspect we would want to also support arbitrary rank tensors on the DL side too. So I would put this off until someone needs it and just let it blow up now.

---

Done